src/components/Dashboard/BalanceTimeline.tsx

```diff
@@ -216,41 +212,35 @@ export default function BalanceTimeline({ height }: { height: number }) {
 
       // Load tap events
       if (showGraph === 'balance') {
-        await querySubgraph(
-          {
-            entity: 'tapEvent',
-            keys: ['netTransferAmount', 'timestamp'],
-            where: projectId
-              ? [
-                  {
-                    key: 'project',
-                    value: projectId.toString(),
-                  },
-                  {
-                    key: 'timestamp',
-                    value: Math.round((now - daysToMillis(duration)) / 1000),
-                    operator: 'gte',
-                  },
-                ]
-              : undefined,
-          },
-          res => {
-            if (res) {
-              newEvents.push(
-                ...res.tapEvents.map(e => {
-                  const event = parseTapEventJson(e)
-                  return {
-                    ...e,
-                    tapped: parseFloat(
-                      parseFloat(fromWad(event.netTransferAmount)).toFixed(4),
-                    ),
-                    timestamp: event.timestamp ?? 0,
-                  }
-                }),
-              )
-            }
-          },
-        )
+        await querySubgraph({
+          entity: 'tapEvent',
```

Comment:
**though (non-blocking)**: not related to this PR at all but something worth considering eventually:
I feel we should aim to move towards defining actual GraphQL queries rather than using a bespoke query builder. I think it'll help make our GraphQL queries easier to maintain and update.

Just something to keep in mind for one day 😊 

Example: https://github.com/tannerlinsley/react-query/blob/master/examples/basic-graphql-request/src/index.js#L48

Reply:
nice idea, i'm open to this!